content/docs/guides/testing/unit.md

```diff
@@ -0,0 +1,964 @@
+    server *ec2.Instance
+}
+
+func createInfrastructure(ctx *pulumi.Context) (*infrastructure, error) {
+    group, err := ec2.NewSecurityGroup(ctx, "web-secgrp", &ec2.SecurityGroupArgs{
+        Ingress: ec2.SecurityGroupIngressArray{
+            ec2.SecurityGroupIngressArgs{
+                Protocol:   pulumi.String("tcp"),
+                FromPort:   pulumi.Int(80),
+                ToPort:     pulumi.Int(80),
+                CidrBlocks: pulumi.StringArray{pulumi.String("0.0.0.0/0")},
+            },
+        },
+    })
+    if err != nil {
+        return nil, err
+    }
+
+    server, err := ec2.NewInstance(ctx, "web-server-www", &ec2.InstanceArgs{
+        InstanceType:   pulumi.String("t2-micro"),
+        SecurityGroups: pulumi.StringArray{group.ID()}, // reference the group object above
+        Ami:            pulumi.String("ami-c55673a0"),  // AMI for us-east-2 (Ohio)
+        Tags:           pulumi.Map{ "Name": pulumi.String("webserver") },
+    })
+    if err != nil {
+        return nil, err
+    }
+
+    return &infrastructure{
+        group:  group,
+        server: server,
+    }, nil
+}
+```
+
+{{% /choosable %}}
+{{% choosable language "csharp" %}}
+**WebserverStack.cs**
+
+```csharp
+using Pulumi;
+using Pulumi.Aws.Ec2;
+using Pulumi.Aws.Ec2.Inputs;
+
+public class WebserverStack : Stack
+{
+    public WebserverStack()
+    {
+        var group = new SecurityGroup("web-secgrp", new SecurityGroupArgs
+        {
+            Ingress =
+            {
+                new SecurityGroupIngressArgs { Protocol = "tcp", FromPort = 80, ToPort = 80, CidrBlocks = { "0.0.0.0/0" } }
+            }
+        });
+
+        var server = new Instance("web-server-www", new InstanceArgs
+        {
+            InstanceType = "t2.micro",
+            SecurityGroups = { group.Name }, // reference the group object above
+            Ami = "ami-c55673a0",            // AMI for us-east-2 (Ohio)
+            Tags = { { "Name", "webserver" }}// name tag
+        });
+    }
+}
+```
+
+{{% /choosable %}}
+
+And then rerun our tests:
+
+{{% choosable language "typescript" %}}
+
+```
+Infrastructure
+    #server
+      ✓ must have a name tag
+      ✓ must not use userData (use an AMI instead)
+    #group
+      ✓ must not open port 22 (SSH) to the Internet
+
+
+  3 passing (454ms)
+```
+
+{{% /choosable %}}
+{{% choosable language "python" %}}
+
+```
+----------------------------------------------------------------------
+Ran 3 tests in 0.022s
+
+OK
+```
+
+{{% /choosable %}}
+{{% choosable language "go" %}}
+
+```
+PASS
+ok  	testing-unit-go	0.704s
+```
+
+{{% /choosable %}}
+{{% choosable language "csharp" %}}
+
+```
+Test Run Successful.
+Total tests: 3
+     Passed: 3
+```
+
+{{% /choosable %}}
+
+All the tests passed!
+
+## Full Example
+
+{{% choosable language "typescript" %}}
+The full code for this guide is available in the examples repository: [Unit Tests in TypeScript](https://github.com/pulumi/examples/tree/master/testing-unit-ts).
```

Comment:
For whatever reason, these links don't look styled correctly (they don't have a different color), so it's not clear that they're links. @cnunciato is this a bug in the chooser?

Reply:
Yeah, that's weird. I could figure out the reason and "fixed" it by adding some random `&nbsp;` at the end.